src/app/app.component.ts

```diff
@@ -81,8 +87,26 @@ export class MyApp {
   openPage(page) {
     // Reset the content nav to have just this page
     // we wouldn't want the back button to show in this scenario
-    this.menuCtrl.close().then(() => {  
-      this.nav.setRoot(page.component, { 'linkParam' : page.param });
+    this.menuCtrl.close().then(() => {
+      let changePageAllowed: boolean = true;
```

Comment:
Do we need this? I think we can just return at the end of each `if`, it might look a bit cleaner.

```
if (...) {
    ...
    return;
}
if (...) {
    ...
    return;
}
this.nav.setRoot(page.component, { 'linkParam' : page.param });
```